app/models/miq_expression/field.rb

```diff
@@ -0,0 +1,46 @@
+class MiqExpression::Field
+  FIELD_REGEX = /
+(?<model_name>[[:upper:]][[:alnum:]]*)
+\.?(?<associations>[a-z_\.]+)*
+-(?<column>[a-z]+(_[a-z]+)*)
+/x
+
+  def self.parse(field)
+    match = FIELD_REGEX.match(field)
+    new(match[:model_name].constantize, match[:associations].to_s.split("."), match[:column])
+  end
+
+  attr_reader :model, :associations, :column
+
+  def initialize(model, associations, column)
+    @model = model
+    @associations = associations
+    @column = column
+  end
+
+  def table_name
+    target.table_name
+  end
+
+  def date?
+    column_type == :date
+  end
+
+  def datetime?
+    column_type == :datetime
+  end
+
+  private
+
+  def target
```

Comment:
@gtanzillo here's the existing strategy for getting the target table name: https://github.com/ManageIQ/manageiq/blob/master/app/models/miq_expression.rb#L1370-L1382. I don't really understand this beyond the simplest case - am I missing something here?